pyproj/_crs.pyx

```diff
@@ -299,13 +276,20 @@ cdef class AreaOfUse:
 cdef class Base:
     def __cinit__(self):
         self.projobj = NULL
+        self.context = NULL
         self.name = "undefined"
 
     def __dealloc__(self):
         """destroy projection definition"""
         if self.projobj != NULL:
             proj_destroy(self.projobj)
-            self.projobj = NULL
+        if self.context != NULL:
+            proj_context_destroy(self.context)
+
+    def initialize_context(self):
```

Comment:
Small comment here: I don't think this method needs to be a public one?

Reply:
That is a horrible method to have public. Definitely needs to be addressed. Thanks for catching that!